### PR TITLE
tests(e2e-tests): Upgrade cypress

### PR DESCRIPTION
**Pull request by akano-baba**
- Updated cypress and cucumber plugin as far as it possible.
- Moved files into new cypress folder structure
- Patched  cucumber plugin to make 'tags' work with new cypress
- tagged @existingFailing tests - pre migration
- tagged @failing tests - require hidden Link config to be removed
- set tests to run in chrome as election crashes out - suspected mem leaks

### CODE REVIEW

File: e2e-tests/package.json [hunk 1]
```diff
@@ -24,12 +25,12 @@
 	},
 	"devDependencies": {
 		"axe-core": "^4.1.3",
-		"cypress": "^7.4.0",
+		"cypress": "^12.3.0",
```
Comment (jonathangoulding) [inline e2e-tests/package.json:28]: cypress open does not work. 

Does the config need mapping to the command ?

File: e2e-tests/cypress.config.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const { defineConfig } = require('cypress')
+const cucumber = require('cypress-cucumber-preprocessor').default
+
+module.exports = (on, config) => {
```
Comment (jonathangoulding) [inline e2e-tests/cypress.config.js:4]: config not used.

**Comment by akano-baba** [comment]
This config is working on my local machine

________________________________

File: e2e-tests/cypress.config.js (hunk 2 above)
Comment (jonathangoulding) [reply e2e-tests/cypress.config.js:4]: the config variable is not used. 

Your IDE / Text editor should be showing this warning. 

No unused vars

Comment (akano-baba) [reply e2e-tests/cypress.config.js:4]: I use vscode, no warning is showing.

Comment (akano-baba) [reply e2e-tests/cypress.config.js:4]: I have removed the config for now.

**Comment by DavidBeale** [comment]
In order to get this to install on my local, I had to bump `cypress-axe` to `1.3.0`. The tests run fine after that.

I also note that this CI build does not even install the e2e-tests deps - I wonder if `e2e-test` should be moved into `packages` and included in the `npm` workspace.

File: e2e-tests/cypress.config.js (hunk 2 above)
Comment (jonathangoulding) [reply e2e-tests/cypress.config.js:4]: the config var is still showing in the PR

**Comment by jonathangoulding** [comment]
> 

did use npm ci or npm i ?

**Comment by DavidBeale** [comment]
> @DavidBeale did you use npm ci or npm i ?

I tried both

File: e2e-tests/cypress.config.js (hunk 2 above)
Comment (akano-baba) [reply e2e-tests/cypress.config.js:4]: I tried pushing with no luck.

**Review by Philip-Philippou1 — DISMISSED** [review]

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**